### ISSUE TITLE
👽 Subscriptions: Returning an array of entities that match with filter criteria from @Args for Subscription resolvers 🤖

### ISSUE THREAD

**Issue by Mingyang-Li** [nestjs/graphql#2385]
### Is there an existing issue that is already proposing this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe it

**👎 Problem 👎:**

1. The current ways of filtering subscriptions in NestJS does not meet my needs for advanced filtering
2. I struggle to build subscription resolvers that take in `@Args`  to filter results

### Describe the solution you'd like

**⭐Ideal solution ⭐:**

Imagine a use-case where the PostgreSQL DB has 2 tables: `User` and `Payment`, where there's a `one-to-many` relationship (One `User` can have many `Payment`, but each `Payment` can only be related to one `User`)
I want to be able to make a GraphQL operation that looks like this:
```gql
Subscription {
  subscribeUsers (
    where: {
      id: { _eq: 'djfdf2387b' },
      createdAt: { _gte: '2022-08-31' }
    },
    orderBy: {
      id: Asc,
      amount: Desc
      createdAt: Asc
    },
    skip: 4,
    take: 6,
  ) {
     id
     firstName
     paymentsCreated {
       id
       amount
       currency
     }
  }
}
```

Or alternatively, similar to what [Hasura](https://github.com/hasura) is capable of doing with their auto-generated subscription resolvers such as:
![image](https://user-images.githubusercontent.com/53138432/188260831-9940846d-a56a-4e5a-b315-b35dc7d2dce7.png)

### What is the motivation / use case for changing the behavior?

**👨‍👦‍👦 Use-case:**
I need to be able to monitor `Payment` table in real-time on the front-end, but I'm not interested in ALL the updates for ALL paynments. Instead, I want to be able to build a `Subscription` resolver that filters payments by the same `@Args` that's being used by the `Query` resolver of `Payment` table in DB

Basically, I'm looking for a way to setup `subscribePayments` so that it uses the same `@Args` as [`payments` resolver](https://github.com/Mingyang-Li/nest-graphql-prisma/blob/75fc1e9824e2cd1e215d5275118a27f11660160a/src/payment/payment.resolver.ts#L32)

You can try run my template project locally by running postgres in docker and play around with the query `@Args`

**Comment by Mingyang-Li** [comment]
PS:
Ideally, for e.g, in our `payment.resolver.ts` file, I want to be able to do something like this:
```ts
import { Args, Mutation, Query, Resolver, Subscription } from '@nestjs/graphql';
import { PubSub } from 'graphql-subscriptions';
import { CreatePaymentArgs } from 'src/payment/dtos/CreatePaymentArgs';
import { UpdatePaymentArgs } from 'src/payment/dtos/UpdatePaymentArgs';
import { Payment } from './dtos/Payment';
import { PaymentFindManyArgs } from './dtos/PaymentFindManyArgs';
import { PaymentService } from './payment.service';

@Resolver(() => Payment)
export class PaymentResolver {
  constructor(private paymentService: PaymentService, private pubSub: PubSub) {}

  @Mutation(() => Payment)
  public async createPayment(
    @Args() args: CreatePaymentArgs,
  ): Promise<Payment> {
    const newPayment = await this.paymentService.create(args);
    this.pubSub.publish('paymentsUpdated', newPayment);
    return newPayment;
  }

  @Mutation(() => Payment)
  public async updatePayment(
    @Args() args: UpdatePaymentArgs,
  ): Promise<Payment> {
    const updatedPayment = await this.paymentService.update(args);
    this.pubSub.publish('paymentsUpdated', updatedPayment);
    return updatedPayment;
  }

  @Query(() => [Payment])
  public async payments(@Args() args: PaymentFindManyArgs): Promise<Payment[]> {
    return this.paymentService.findMany(args);
  }

  @Subscription(() => [Payment])
  public async subscribePayments(@Args() args: PaymentFindManyArgs) {
    // whatever code we need to return an array of payments that match the filter criteria from @Args
  }
}
```

How feasible is this on considering your current roadmap?

**Comment by kamilmysliwiec** [comment]
Thanks for your suggestion!

This has been discussed in the past and we decided to not implement it in the foreseeable future.

If you think your request could live outside Nest's scope, we'd encourage you to collaborate with the [community](https://discord.gg/G7Qnnhy) on publishing it as an open source package (or an extension of this package).
